### DISCUSSION TITLE
Analyze image with unequal voxel sizes in each direction

### DISCUSSION THREAD

**Discussion by chunbrian16** [PMEAL/porespy#684] [Q&A]
Hi, I’m try to analyze some z stack images of collagen fibers taken from a confocal. My voxel size (x y z) is 0.2678um x 0.2678um x 0.5um. I’m wondering how I should calibrate my image before running the code so I can get a correct pore size distribution?

Thank you!

**Comment by jgostick** [comment]
PoreSpy does not support unequal voxel lengths, and doing so would be a huge pain...but luckily there is a workaround for you. Use ``scipy.ndimage.zoom`` which accepts direction-dependent scale factors, so you can either zoom "down" in the x and y direction (and lose resolution) or zoom up in the z direciton and have the function interpolate for you.  The fact that the resolutions are not equal ratios of each other is a bit of a pain, since this means you'll need to use some sort of higher order interpolation (as opposed to 0 order which is just 'nearest neighbor').  So if your image is not too big and you can afford to scale up, then it's be something like:

``im = scipy.ndimage.zoom(im.astype(float), zoom=[1, 1, F], order=3)``

where ``F`` is 0.5/0.2768.  Now all your voxels will be of size 0.2678 um.  You will need to post-process the zoomed image though to turn it back into a boolean, since the zoom will convert it to floats and cause some noise near the edges.  Good luck!

**Comment by chunbrian16** [comment]
Thanks for your detailed explanation @jgostick ! Just want to confirm one thing before proceeding: If I resize the image, then I use this resized image to run `metrics.pore_size_distribution`, how shall I resize my x-axis (pore size in radius) of the distribution plot back to the true dimension?
Thanks in advance!

**Reply by jgostick** [reply]
You don't have to resize anything.  You're actually 'fixing' the flawed image by making the voxels equal length in all directions.  Now ALL analyze you do will be valid.  You basically just say "now my voxels are all 0.2678 um in all directions.
